src/ke.cpp

```diff
@@ -227,8 +227,8 @@ KeSetSystemAffinityThreadEx(KAFFINITY affinity)
     _usersim_thread_affinity.Group = old_affinity.Group;
     _usersim_thread_affinity.Mask = affinity;
     if (KeGetCurrentIrql() < DISPATCH_LEVEL && SetThreadAffinityMask(GetCurrentThread(), affinity) == 0) {
-        unsigned long error = GetLastError();
-        KeBugCheckEx(0, error, 0, 0, 0);
+        _usersim_thread_affinity = old_affinity;

```

Comment:
Is the behavior correct if we happen to be at dispatch level?

Reply:
It appears to be. If IRQL ==  DISPATCH_LEVEL, then the affinity is set when IRQL is lowered.

Reply:
Actually, the behavior is wrong. It should test if the affinity is valid before setting the thread_local value otherwise it will fail when it lowers the IRQL.